src/Giraffe/HttpHandlers.fs

```diff
@@ -351,31 +351,34 @@ let renderHtml (document : XmlNode) : HttpHandler =
 /// Example:
 /// let negotiationRules = dict [ "application/json", json; "application/xml" , xml ]
 /// `json` and `xml` are both the respective default HttpHandler functions in this example.
-let negotiateWith (negotiationRules    : IDictionary<string, obj -> HttpHandler>)
+let negotiateWith (negotiationRules    : #IDictionary<string, obj -> HttpHandler>)
                   (unacceptableHandler : HttpHandler)
+                  (defaultMimeType     : string)
```

Comment:
Personally I think this is redundant/duplicate information. The default mime type is only relevant when a client has no preference whatsoever, in that case the server already has a dict of supported mime types and a user has to pass it into `negotiateWith`, so they might as well set the dict in preferred order (even though it doesn't matter to the client).

Reply:
As in what I am trying to say is that I think that a default mime type as a concept doesn't make sense, because if a client doesn't care then they don't care and so why should the server care which one to serve. The response is only for the client and if the client doesn't care, then the server might as well just pick a random one and instead of random one I think taking the first from the list has at least an expected behaviour.

Reply:
yes the problem is: there is no first. If you don't care and random is what you want then we can revert that

Reply:
You mean a dict has no concept of first? Yes, I'd be happy to revert and pick a random, I'd also be happy to use a collection type which has a first/last. I think the only things I'd want is that if I make the same request from a client which has no Accept header that the server will at least always return a response in the same mimeType.

Reply:
done

Reply:
yes it's still predictable. In the dictionary case it will select first in list. In map case it will select the one with smallest key (alphabetical order)

Reply:
yeah that's what I was trying to get at earlier, it will be consistent but different depending on dict/map type